addons/mail/static/src/models/thread/thread.js

```diff
@@ -367,15 +396,58 @@ function factory(dependencies) {
         }
 
         /**
+         * Opens a chat between the current partner and the given partner.
+         * If a chat is not appropriate, the form view is opened instead.
+         *
+         * If a chat already exists between those partners it is reused.
+         * Otherwise a new chat is created.
+         *
          * @static
+         * @param {Object} param0
+         * @param {integer} param0.partnerId
+         * @param {Object} [param0.threadOpenOptions]
          */
-        static openNewMessage() {
-            const discuss = this.env.messaging.discuss;
-            if (discuss.isOpen) {
-                discuss.openNewMessage();
-            } else {
-                this.env.messaging.chatWindowManager.openNewMessage();
+        static async openChat({ partnerId, threadOpenOptions } = {}) {
```

Comment:
I would change signature of function with 2 position params: 1st is for data, and 2nd is for options:
```js
static async openChat({ partnerId }, options);
```

Reply:
ok, done